src/rules/no_node_globals.rs

```diff
@@ -0,0 +1,308 @@
+        changes: vec![change],
+      }],
+    );
+  }
+}
+
+impl Handler for NoNodeGlobalsHandler {
+  fn ident(&mut self, id: &ast_view::Ident, ctx: &mut Context) {
+    if !NODE_GLOBALS.contains_key(id.sym()) {
+      return;
+    }
+    if id.ctxt() == ctx.unresolved_ctxt() {
+      self.add_diagnostic(ctx, id.range(), NODE_GLOBALS[id.sym()]);
+    }
+  }
+
+  fn import_decl(&mut self, imp: &ast_view::ImportDecl, _ctx: &mut Context) {
+    self.most_recent_import_range = Some(imp.range());
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn valid() {
+    assert_lint_ok! {
+      NoNodeGlobals,
+      "import process from 'node:process';\nconst a = process.env;",
+      "const process = { env: {} };\nconst a = process.env;",
+      "import { Buffer } from 'node:buffer';\nconst b = Buffer;",
+      "const Buffer = {};\nconst b = Buffer;",
+      "const global = globalThis;\nconst c = global;",
+      "const setImmediate = () => {};\nconst d = setImmediate;",
+      "const clearImmediate = () => {};\nconst e = clearImmediate;",
+    }
+  }
+
+  #[test]
+  fn invalid() {
+    assert_lint_err! {
+      NoNodeGlobals,
+      "import a from 'b';\nconst e = process.env;": [
+        {
+          col: 10,
+          line: 2,
+          message: MESSAGE,
+          hint: IMPORT_HINT,
+          fix: (
+            IMPORT_FIX_DESC,
+            "import a from 'b';\nimport process from \"node:process\";\nconst e = process.env;"
+          ),
+        }
+      ],
+      "const a = process;": [
+        {
+          col: 10,
+          line: 1,
+          message: MESSAGE,
+          hint: IMPORT_HINT,
+          fix: (
+            IMPORT_FIX_DESC,
+            "import process from \"node:process\";\nconst a = process;"
+          ),
+        }
+      ],
+      "const b = Buffer;": [
+        {
+          col: 10,
+          line: 1,
+          message: MESSAGE,
+          hint: IMPORT_HINT,
+          fix: (
+            IMPORT_FIX_DESC,
+            "import { Buffer } from \"node:buffer\";\nconst b = Buffer;"
+          ),
+        }
+      ],
+      "const c = global;": [
+        {
+          col: 10,
+          line: 1,
+          message: MESSAGE,
+          hint: REPLACE_HINT,
+          fix: (
+            REPLACE_FIX_DESC,
+            "const c = globalThis;"
+          ),
+        }
+      ],
+      "const d = setImmediate;": [
+        {
+          col: 10,
+          line: 1,
+          message: MESSAGE,
+          hint: IMPORT_HINT,
+          fix: (
+            IMPORT_FIX_DESC,
+            "import { setImmediate } from \"node:timers\";\nconst d = setImmediate;"
+          ),
+        }
+      ],
+      "const e = clearImmediate;": [
+        {
+          col: 10,
+          line: 1,
+          message: MESSAGE,
+          hint: IMPORT_HINT,
+          fix: (
+            IMPORT_FIX_DESC,
+            "import { clearImmediate } from \"node:timers\";\nconst e = clearImmediate;"
```

Comment:
Can you add a test for when there's a copyright comment at the top of the file and no imports?